lib/iwdp.js

```diff
@@ -0,0 +1,81 @@
+import { exec, SubProcess } from 'teen_process';
+import EventEmitter from 'events';
+import logger from './logger';
+import request from 'request-promise';
+import { retryInterval } from 'asyncbox';
+
+const IWDP_CMD = 'ios_webkit_debug_proxy';
+const MAX_RETRIES = 10;
+
+class IWDP extends EventEmitter {
+
+  constructor (udid = null) {
+    super();
+    this.udid = udid;
+    this.attempts = 0;
+    this.port = 27753;
+    this.process = this.createIWDPProcess(); 
+    this.endpoint = `http://localhost:${this.port}`;
+  }
+
+  createIWDPProcess () {
+    let process = new SubProcess(IWDP_CMD, ['--config', `${this.udid}:${this.port},:${this.port + 1}-${this.port + 101}`]); // (see https://github.com/google/ios-webkit-debug-proxy for reference)// If the process exits and the exit wasn't requested by the API, restart it
+    process.on('exit', () => this.onExit());
+    return process;
+  }
+
+  async onExit () {
+    if (!this.exitRequested) {
+      this.process = this.createIWDPProcess();
+      await this.start(); 
+    }
+  }
+
+  async start () {
+    if (++this.attempts > MAX_RETRIES) {
```

Comment:
Still a little confused here. It seems that we could encounter an exit event in 3 situations:

1. While we're waiting for IWDP to start initially
2. When we're intentionally stopping IWDP
3. When IWDP exits at some other point in the middle of use.

We're already handling case 1 in the retryInterval (if IWDP doesn't start, we won't be able to request to it)

In case 2, we don't care about hearing when it exits.

In case 3, why would we want to limit the number of times we restart IWDP in the middle of a test?